src/components/PostCard/PostCard.jsx

```diff
@@ -51,14 +120,89 @@ const PostCard = ({ postData }) => {
             • {day - 0} {month}
           </p>
         </div>
+        <div
+          className={`ml-auto relative ${
+            username === userProfile.username ? "visible" : "hidden"
+          }`}
+        >
+          <MoreVertIcon
+            sx={{ width: "2rem", height: "2rem", padding: "0.2rem" }}
+            className="hover:bg-slate-200 rounded-full cursor-pointer"
+            onClick={tooltipHandler}
+          />
+          <div
+            className={`border rounded-sm absolute right-2 bg-white ${isTooltipVisible}`}
+          >
+            <div
+              className="flex m-1 border-b gap-2 hover:bg-slate-200 cursor-pointer rounded-sm p-1"
+              onClick={() => editPostHandler()}
```

Comment:
why are you invoking the callback here? Simply pass it like `onClick={editPostHandler}`. It will work.